Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SingleCrystalDiffuseReduction.py

```diff
@@ -197,134 +215,112 @@ def PyExec(self):
+            CropWorkspace(InputWorkspace='__flux',OutputWorkspace='__flux',XMin=self.XMin,XMax=self.XMax)
             for spectrumNumber in range(mtd['__flux'].getNumberHistograms()):
                 Y = mtd['__flux'].readY(spectrumNumber)
                 mtd['__flux'].setY(spectrumNumber,(Y-Y.min())/(Y.max()-Y.min()))
 
         if _background:
-            Load(Filename=self.getProperty("Background").value,
-                 OutputWorkspace='__bkg',
-                 FilterByTofMin=self.getProperty("FilterByTofMin").value,
-                 FilterByTofMax=self.getProperty("FilterByTofMax").value)
-            if _load_inst:
-                LoadInstrument(Workspace='__bkg', Filename=self.getProperty("LoadInstrument").value, RewriteSpectraMap=False)
-            if _detcal:
-                LoadIsawDetCal(InputWorkspace='__bkg', Filename=self.getProperty("DetCal").value)
-            MaskDetectors(Workspace='__bkg',MaskedWorkspace='__sa')
-            ConvertUnits(InputWorkspace='__bkg',OutputWorkspace='__bkg',Target='Momentum')
-            CropWorkspace(InputWorkspace='__bkg',OutputWorkspace='__bkg',XMin=XMin,XMax=XMax)
-
-        progress = Progress(self, 0.0, 1.0, len(UBList)*len(self.getProperty("Filename").value))
-
-        for run in self.getProperty("Filename").value:
-            logger.notice("Working on " + run)
+            self.load_file_and_apply(self.getProperty("Background").value, '__bkg', 0)
 
-            Load(Filename=run,
-                 OutputWorkspace='__run',
-                 FilterByTofMin=self.getProperty("FilterByTofMin").value,
-                 FilterByTofMax=self.getProperty("FilterByTofMax").value)
-            if _load_inst:
-                LoadInstrument(Workspace='__run', Filename=self.getProperty("LoadInstrument").value, RewriteSpectraMap=False)
-            if _detcal:
-                LoadIsawDetCal(InputWorkspace='__run', Filename=self.getProperty("DetCal").value)
-            MaskDetectors(Workspace='__run',MaskedWorkspace='__sa')
-            ConvertUnits(InputWorkspace='__run',OutputWorkspace='__run',Target='Momentum')
-            CropWorkspace(InputWorkspace='__run',OutputWorkspace='__run',XMin=XMin,XMax=XMax)
+        progress = Progress(self, 0.0, 1.0, len(self.getProperty("Filename").value))
 
-            if self.getProperty('SetGoniometer').value:
-                SetGoniometer(Workspace='__run',
-                              Goniometers=self.getProperty('Goniometers').value,
-                              Axis0=self.getProperty('Axis0').value,
-                              Axis1=self.getProperty('Axis1').value,
-                              Axis2=self.getProperty('Axis2').value)
+        for n, run in enumerate(self.getProperty("Filename").value):
+            logger.notice("Working on " + run)
+
+            self.load_file_and_apply(run, '__run', _offsets[n])
+            LoadIsawUB('__run', _UB[n])
+
+            MinValues, MaxValues = ConvertToMDMinMaxGlobal(InputWorkspace='__run',
```

Comment:
I just realized that `MinValues` are all `-self.XMax` and `MaxValues` are all `self.XMax`

Reply:
So it is. I will change this.